interop-tests/src/bin/ping.rs

```diff
@@ -166,18 +167,18 @@ fn secure_channel_protocol_from_env<C: AsyncRead + AsyncWrite + Unpin + Send + '
 ) -> Result<
     MapOutboundUpgrade<
         MapInboundUpgrade<
-            Either<noise::NoiseAuthenticated<XX, X25519Spec, ()>, tls::Config>,
+            EitherUpgrade<noise::NoiseAuthenticated<XX, X25519Spec, ()>, tls::Config>,
```

Comment:
Which one is preferred here @thomaseizinger?

Reply:
`master` requires `Either` Marco. `EitherUpgrade` was removed on https://github.com/libp2p/rust-libp2p/pull/3339

Reply:
If you leave all the "either" code as it is on master (i..e remove all this diff), then it should work. Let me know if you run into trouble!